src/dmd/root/array.h

```diff
@@ -16,12 +16,12 @@ template <typename TYPE>
 struct Array
 {
     d_size_t dim;
-    TYPE *data;
 
   private:
     Array(const Array&);
 
     d_size_t allocdim;
+    TYPE *data;
```

Comment:
This should be a `DArray`

Reply:
That would break users of it, which I don't have the code for. So I'll let those users make such changes.

Reply:
Leaving it as a pointer would break things anyway as Arras type sizes would differ between C++ and D.

Reply:
It took me a while to work out that you left in `allocdim`, so the sizes do indeed match in both C++ and D, thus nothing breaks.

If you are really that uncomfortable making identical changes to both the C++ and D implementation of Array, then I can fix things up in a follow-up.  As most of Array is extern(C++) anyway, we could even remove the inline implementations in the headers.